### PR TITLE
feat: initial API

### PR DESCRIPTION
**Pull request by artursvonda**
Exposes `styled`, `css`, `createGlobalStyle` and `keyframes` functions from `styled-components`.

Adds on top a few declarative helpers like `modifier` and `variants`. And exposes `generateHelpers` function that generates declarative theme helpers. And convenience helper to remove extranous props from DOM using `excludeProps`.

### CODE REVIEW

File: vite.config.ts [hunk 1]
```diff
@@ -1,16 +1,33 @@
 import { resolve } from "path";
 import { defineConfig } from "vitest/config";
 import dts from "vite-plugin-dts";
+import packageJson from "./package.json";
 
 export default defineConfig({
```
Comment (rupert-mckay) [inline vite.config.ts:6]: 🟢 Let's raise a ticket to back-port these changes into the template

File: .eslintrc.json [hunk 2]
```diff
@@ -5,8 +5,18 @@
...
-    "@lokalise/lokalise-frontend/jest",
-    "@lokalise/lokalise-frontend/typescript"
+    "@lokalise/eslint-config-lokalise-frontend/core",
+    "@lokalise/eslint-config-lokalise-frontend/jest",
+    "@lokalise/eslint-config-lokalise-frontend/typescript"
+  ],
+  "overrides": [
+    {
+      "files": ["*.cjs"],
+      // Fall back to default parser for non-ts files
```
Comment (rupert-mckay) [inline .eslintrc.json:15]: ❓ Was there an problem if this override isn't provided? If so let's raise a ticket to incorporate back into the eslint config.

**Review by rupert-mckay — APPROVED** [review]
Can't honestly say I understood it all, but the thorough tests give me confidence 😄

File: .eslintrc.json (hunk 2 above)
Comment (artursvonda) [reply .eslintrc.json]: The issue was that I included only `src` folder in tsconfig so anything outside it wasn't getting handled by tsconfig and, by extension, by ts resolver / loader / whatever. In other words, eslint couldn't understand files outside src folder. I adjusted to include only `src` when compiling d.ts files though vite.

File: vite.config.ts (hunk 1 above)
Comment (artursvonda) [reply vite.config.ts:6]: Will do

Comment (artursvonda) [reply vite.config.ts:6]: https://github.com/lokalise/npm-package-template/pull/32/files

**Review by mascardoso — APPROVED** [review]

**Review by sangdth — APPROVED** [review]

**Review by arthuracs — APPROVED** [review]

**Pull request merged**